### PR TITLE
feat(fs): `Deno.FsFile.{isTerminal,setRaw}()`

### LINKED ISSUE #22229: suggestion: implement `Deno.FsFile.isTerminal()`
In the case that someone does something like `Deno.open("/dev/tty6")`.

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #22229.

### CODE REVIEW

File: cli/tsc/dts/lib.deno.ns.d.ts [hunk 1]
```diff
@@ -2666,6 +2666,16 @@ declare namespace Deno {
...
+    /**
+     * Checks if the file resource is a TTY (terminal).
+     *
+     * ```ts
+     * // This example is system and context specific
+     * using file = await Deno.open("/dev/tty6");
+     * file.isTerminal(); // true
+     * ```
+     */
+    isTerminal(): boolean;
```
Comment (bartlomieju) [inline cli/tsc/dts/lib.deno.ns.d.ts:2678]: We should mark it as unstable for now so we can land it in the next patch release

**Comment by iuioiua** [comment]
How can we test `setRaw()`?

**Comment by bartlomieju** [comment]
> How can we test `setRaw()`?

It should be an integration test, you should be able to find examples in `run_tests.rs`

Please rebase.

File: runtime/js/40_tty.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ function isatty(rid) {
   internals.warnOnDeprecatedApi(
     "Deno.isatty()",
     new Error().stack,
-    "Use `Deno.stdin.isTerminal()`, `Deno.stdout.isTerminal()` or `Deno.stderr.isTerminal()` instead.",
+    "Use `Deno.FsFile.isTerminal()`, `Deno.stdin.isTerminal()`, `Deno.stdout.isTerminal()` or `Deno.stderr.isTerminal()` instead.",
```
Comment (bartlomieju) [inline runtime/js/40_tty.js:19]: Can you move `FsFile` to be the last - it's gonna be used rarely compared to the other ones

**Review by bartlomieju — COMMENTED** [review]
Looks good, @dsherret WDYT?

**Review by dsherret — APPROVED** [review]
LGTM too

**Pull request merged**